### PR TITLE
Write output to generate_workspace.bzl by default, instead of WORKSPACE and BUILD files

### PR DESCRIPTION
**Pull request by kchodorow**
This splits the output writer into old & new versions. By default, `generate_workspace` will write generate_workspace.bzl, but users can specify `--direct-to-ws` to get the old behavior.

See BzlWriter.java#L54 for new usage info.

### CODE REVIEW

File: generate_workspace/src/main/java/com/google/devtools/build/workspace/output/BzlWriter.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    }
+
+    outputStream.append("\n\n");
+
+    outputStream.println("def generated_java_libraries():");
+    if (rules.isEmpty()) {
+      outputStream.println("  pass\n");
+    }
+    for (Rule rule : rules) {
+      outputStream.println("  native.java_library(");
```
Comment (katre) [inline generate_workspace/src/main/java/com/google/devtools/build/workspace/output/BzlWriter.java]: Move this into a formatJavaLibrary method, similar to formatMavenJar.

File: generate_workspace/src/main/java/com/google/devtools/build/workspace/output/WorkspaceWriter.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  public void writeWorkspace(PrintStream outputStream, List<String> sources,
+      Collection<Rule> rules) {
+    writeHeader(outputStream, sources);
+    for (Rule rule : rules) {
+      outputStream.println(formatMavenJar(rule) + "\n");
+    }
+  }
+
+  @Override
+  public String formatMavenJar(Rule rule) {
```
Comment (katre) [inline generate_workspace/src/main/java/com/google/devtools/build/workspace/output/WorkspaceWriter.java]: Should this move back to the AbstractWriter? I can't actually tell how it's different from the version in BzlWriter.

**Review by katre — COMMENTED** [review]
Is it possible to add a shell-based integration test for this to ensure the generated rules work with Bazel? It'd also serve as an example of how to use the generator.

File: generate_workspace/src/main/java/com/google/devtools/build/workspace/output/WorkspaceWriter.java (hunk 2 above)
Comment (kchodorow) [reply generate_workspace/src/main/java/com/google/devtools/build/workspace/output/WorkspaceWriter.java]: It's prefixed by "native." and indented an extra 2 spaces :-/

Comment (katre) [reply generate_workspace/src/main/java/com/google/devtools/build/workspace/output/WorkspaceWriter.java]: The logic is complicated enough I think it's still worth unifying, just passing in the initial indent and the rule name.

File: generate_workspace/src/main/java/com/google/devtools/build/workspace/output/BzlWriter.java (hunk 1 above)
Comment (kchodorow) [reply generate_workspace/src/main/java/com/google/devtools/build/workspace/output/BzlWriter.java]: Done.

**Comment by kchodorow** [comment]
Wow, it is super annoying writing shell tests without unittest.bash. Working on that...

**Comment by kchodorow** [comment]
That's a no on the shell tests. It's too hard to get a copy of Bazel (sadly) and generate_workspace contact Maven Central, which seems suboptimal.

I added more info to the README and Michelle's currently got a CL in progress documenting this.

**Review by katre — APPROVED** [review]

**Comment by katre** [comment]
Too bad on the shell tests, but the README is definitely helpful.

**Pull request merged**